### PR TITLE
Run generator in multiprocess

### PR DESCRIPTION
**Pull request by SergeyRyabinin**
*Issue #, if available:*
Generation step takes some time
*Description of changes:*
Run generator in multiple processes in parallel.
Move complicated cmake logic to python.
*Check all that applies:*
- [ ] Did a review by yourself.
- [ ] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [ ] Checked if this PR is a breaking (APIs have been changed) change.
- [ ] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [ ] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [ ] Linux
- [ ] Windows
- [ ] Android
- [ ] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/run_code_generation.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        if "." in key:
+            key = "-".join(reversed(key.split(".")))  # just replicating existing legacy behavior
+        if ";" in key:
+            key = key.replace(";", "-")  # just in case... just replicating existing legacy behavior
+
+        service_name_to_model_filename[key] = model_file_date[0]
+        if key == "s3":
+            service_name_to_model_filename["s3-crt"] = model_file_date[0]
+
+    # for key, val in service_name_to_model_filename.items():
```
Comment (sbiscigl) [inline scripts/run_code_generation.py:74]: nit pick: we could remove the commented out code

**Review by sbiscigl — APPROVED** [review]

File: scripts/run_code_generation.py (hunk 1 above)
Comment (SergeyRyabinin) [reply scripts/run_code_generation.py]: done

File: scripts/run_code_generation.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        if not os.path.exists(output_filename) or os.path.getsize(output_filename) < 4:
+            raise RuntimeError(f"Code of {service_name} generation failure: "
+                               f"Code generator did not generate an output archive (and did not report failure!)")
+
+    if output_filename != "STDOUT":
+        output_zip_file = output_filename
+    else:
+        output_zip_file = process.stdout
+        if not output_zip_file or len(output_zip_file) < 4:
+            raise RuntimeError(f"Cdde of {service_name} generation failure: "
```
Comment (sdavtaker) [inline scripts/run_code_generation.py:95]: typo in error

File: scripts/run_code_generation.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+    return service_name, 0
+
+
+def generate_defaults(models_filepath: str,
+                      generator_filepath: str,
+                      output_dir: str,
+                      tmp_dir: str,
+                      kwargs):
+
+    if not kwargs.get("language-binding"):
```
Comment (sdavtaker) [inline scripts/run_code_generation.py:125]: what are the options here?

File: scripts/run_code_generation.py [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+    parser.add_argument("--path_to_api_definitions", action="store")
+    parser.add_argument("--path_to_generator", action="store")
+    parser.add_argument("--prepare_tools", help="Just build the generator", action="store_true")
+
+    parser.add_argument("--list_all", help="Lists all available SDKs for generation.", action="store_true")
+    # raw generator params to pass through
+    parser.add_argument("--license-text", action="store")
+    parser.add_argument("--namespace", action="store")
+    parser.add_argument("--standalone",
+                        help="Build custom client as a separete package, with prebuilt C++ SDK as dependency",
```
Comment (sdavtaker) [inline scripts/run_code_generation.py:205]: Typo in 'separate'

**Review by sdavtaker — CHANGES_REQUESTED** [review]
Just a few typos in the output. Marked them in the line-comments.

**Review by sdavtaker — APPROVED** [review]

**Pull request merged**